### PR TITLE
fix for windows path to show in project tree across platforms correctly

### PR DESCRIPTION
**Pull request by udeeshagautam**
This PR 
Fixes https://github.com/microsoft/azuredatastudio/issues/10986
Converting the relative path to platform safe path before using in uri. 
Opened a separate bug for MAC --> SSDT.

Fixes https://github.com/microsoft/azuredatastudio/issues/11028 
Adding one more change to allow retry on projects that did not load

### CODE REVIEW

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -85,11 +85,19 @@ export class ProjectsController {
...
 			}
 		}
 
-		this.refreshProjectsTree();
+		try {
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			// if the project didnt load - remove it from the list of open projects
+			this.projects = this.projects.filter((e) => { return e !== newProject; });
```
Comment (Benjin) [inline extensions/sql-database-projects/src/controllers/projectController.ts:98]: I think VS/SSDT shows the project in the tree, but with a "failed to load" message.  I like the idea of doing that here, too, but maybe that's worth doing in a separate task.

File: extensions/sql-database-projects/src/models/project.ts [hunk 2]
```diff
@@ -250,7 +250,8 @@ export class Project {
 	}
 
 	public createProjectEntry(relativePath: string, entryType: EntryType): ProjectEntry {
-		return new ProjectEntry(Uri.file(path.join(this.projectFolderPath, relativePath)), relativePath, entryType);
+		let platformSafeRelativePath = utils.getPlatformSafeFileEntryPath(relativePath);
+		return new ProjectEntry(Uri.file(path.join(this.projectFolderPath, platformSafeRelativePath)), relativePath, entryType);
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/project.ts:254]: Can you also update DatabaseReferenceProjectEntries to have platform safe path? Or I can do it in another PR after this is checked in :) It also has the same problem that it doesn't parse the path correctly on Mac for Windows paths.

Comment (udeeshagautam) [reply extensions/sql-database-projects/src/models/project.ts:254]: will do it and send another itr.

File: extensions/sql-database-projects/src/controllers/projectController.ts (hunk 1 above)
Comment (udeeshagautam) [reply extensions/sql-database-projects/src/controllers/projectController.ts:98]: yeah we can do that with a reload context menu button. Let's go with this for now t unblock and do that as a seperate thing.

Comment (Benjin) [reply extensions/sql-database-projects/src/controllers/projectController.ts:98]: Sounds good!  Created as #11047

**Pull request merged**